src/sidebar/services/test/autosave-test.js

```diff
@@ -0,0 +1,94 @@
+import * as annotationFixtures from '../../test/annotation-fixtures';
+import createFakeStore from '../../test/fake-redux-store';
+import { waitFor } from '../../../test-util/wait';
+
+import autosaveService from '../autosave';
+import { $imports } from '../autosave';
+
+describe('autosaveService', () => {
+  let fakeAnnotationsService;
+  let fakeNewHighlights;
+  let fakeRetryPromiseOperation;
+  let fakeStore;
+
+  beforeEach(() => {
+    fakeAnnotationsService = { save: sinon.stub().resolves() };
+    fakeNewHighlights = sinon.stub().returns([]);
+    fakeRetryPromiseOperation = sinon.stub().callsFake(callback => callback());
+    fakeStore = createFakeStore(
+      { annotations: [] },
```

Comment:
Is there a need to actually have any data in the fake store, since you're always using the `newHighlights` method to get the data? If not, I'd just leave the fake store with an empty state.